homeassistant/components/shelly/climate.py

```diff
@@ -316,6 +316,21 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         if (current_temp := kwargs.get(ATTR_TEMPERATURE)) is None:
             return
+
+        # Shelly TRV accepts target_t in Fahrenheit or Celsius, but you must
+        # send the units that the device expects
+        if self.block is not None and self.block.channel is not None:
+            therm = self.coordinator.device.settings["thermostats"][
+                int(self.block.channel)
+            ]
+            LOGGER.debug("Themostat settings: %s", therm)
+            if therm.get("target_t", {}).get("units", "C") == "F":
```

Comment:
This was the best way I found to pull the real setting from the TRV... If there is a better way, i'd be glad to hear it. Ideally we'd already have this parsed out and in some attribute of the class already without having to poke at it here. Also, we can reach here while entities are still loading (i think) and it's probably not ideal to even try to set the temperatures if we know we're not fully initialized., but that's not dealt with in this PR.